### ISSUE TITLE
[JENKINS-23256] Job hangs trying to resolve Perforce emails, in a non-perforce job

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/mailer-plugin#508]
<p>We upgraded from Jenkins 1.445 to 1.565, and we have a number of jobs configured for subversion, and some configured for perforce.</p>

<p>This one job in particular, failed (after I had to kill the p4 executable on the build agent, and then ultimately the build agent itself) when it tried to generate an email list to send emails out. It appears to be attempting to get users emails from perforce, even though this one job in particular is in no way configured to use Perforce.</p>

```
ERROR: Publisher hudson.tasks.Mailer aborted due to exception
java.lang.NullPointerException
	at hudson.model.Slave.createLauncher(Slave.java:359)
	at hudson.plugins.perforce.PerforceMailResolver.findPerforceMailAddressFor(PerforceMailResolver.java:74)
	at hudson.plugins.perforce.PerforceMailResolver.findMailAddressFor(PerforceMailResolver.java:25)
	at hudson.tasks.MailAddressResolver.resolve(MailAddressResolver.java:112)
	at hudson.tasks.Mailer$UserProperty.getAddress(Mailer.java:547)
	at hudson.tasks.MailSender.buildCulpritList(MailSender.java:407)
	at hudson.tasks.MailSender.createEmptyMail(MailSender.java:367)
	at hudson.tasks.MailSender.createBackToNormalMail(MailSender.java:176)
	at hudson.tasks.MailSender.getMail(MailSender.java:167)
	at hudson.tasks.MailSender.execute(MailSender.java:101)
	at hudson.tasks.Mailer.perform(Mailer.java:137)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:745)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:709)
	at hudson.model.Build$BuildExecution.post2(Build.java:182)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:658)
	at hudson.model.Run.execute(Run.java:1731)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:231)

```

<p>Only once I had disabled the Perforce plugin entirely, would the job complete without hanging.</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=davyboyhayes">davyboyhayes</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-23256" target="_blank">Job hangs trying to resolve Perforce emails, in a non-perforce job</a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Major
<li><b>component(s)</b>: mailer-plugin, p4-plugin
<li><b>label(s)</b>: email-ext, mailer, perforce
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 0
<li><b>watchers</b>: 5
<li><b>imported</b>: 20251211-223321
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>We upgraded from Jenkins 1.445 to 1.565, and we have a number of jobs configured for subversion, and some configured for perforce.</p>

<p>This one job in particular, failed (after I had to kill the p4 executable on the build agent, and then ultimately the build agent itself) when it tried to generate an email list to send emails out. It appears to be attempting to get users emails from perforce, even though this one job in particular is in no way configured to use Perforce.</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>ERROR: Publisher hudson.tasks.Mailer aborted due to exception
java.lang.NullPointerException
	at hudson.model.Slave.createLauncher(Slave.java:359)
	at hudson.plugins.perforce.PerforceMailResolver.findPerforceMailAddressFor(PerforceMailResolver.java:74)
	at hudson.plugins.perforce.PerforceMailResolver.findMailAddressFor(PerforceMailResolver.java:25)
	at hudson.tasks.MailAddressResolver.resolve(MailAddressResolver.java:112)
	at hudson.tasks.Mailer$UserProperty.getAddress(Mailer.java:547)
	at hudson.tasks.MailSender.buildCulpritList(MailSender.java:407)
	at hudson.tasks.MailSender.createEmptyMail(MailSender.java:367)
	at hudson.tasks.MailSender.createBackToNormalMail(MailSender.java:176)
	at hudson.tasks.MailSender.getMail(MailSender.java:167)
	at hudson.tasks.MailSender.execute(MailSender.java:101)
	at hudson.tasks.Mailer.perform(Mailer.java:137)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:745)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:709)
	at hudson.model.Build$BuildExecution.post2(Build.java:182)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:658)
	at hudson.model.Run.execute(Run.java:1731)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:88)
	at hudson.model.Executor.run(Executor.java:231)
</pre>
</div></div>

<p>Only once I had disabled the Perforce plugin entirely, would the job complete without hanging.</p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
email-ext 2.38<br/>
perforce 1.3.27<br/>
jenkins 1.565<br/>
mailer 1.8
```
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-23256?focusedId=202675&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-202675">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Kohsuke Kawaguchi
Path:
 src/main/java/hudson/tasks/MailAddressResolver.java
<a href="http://jenkins-ci.org/commit/mailer-plugin/c5497e08f35e18ae5a37ace9139a651363d8658e" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/mailer-plugin/c5497e08f35e18ae5a37ace9139a651363d8658e</a>
Log:
  <a href="https://issues.jenkins.io/browse/JENKINS-23256" title="Job hangs trying to resolve Perforce emails, in a non-perforce job" class="issue-link" data-issue-key="JENKINS-23256">JENKINS-23256</a></p>

<p>Don't let a problem in MailAddressResolver affect the innocent caller</p>


<p>Compare: <a href="https://github.com/jenkinsci/mailer-plugin/compare/38c84e2320ee...c5497e08f35e" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/mailer-plugin/compare/38c84e2320ee...c5497e08f35e</a></p>

</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Kohsuke Kawaguchi<br/>
Path:<br/>
 src/main/java/hudson/tasks/MailAddressResolver.java<br/>
<a href="http://jenkins-ci.org/commit/mailer-plugin/c5497e08f35e18ae5a37ace9139a651363d8658e" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/mailer-plugin/c5497e08f35e18ae5a37ace9139a651363d8658e</a><br/>
Log:<br/>
  <a href="https://issue-redirect.jenkins.io/issue/23256" title="Job hangs trying to resolve Perforce emails, in a non-perforce job" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/23256">JENKINS-23256</a>">JENKINS-23256</a></p>

<p>Don't let a problem in MailAddressResolver affect the innocent caller</p>


<p>Compare: <a href="https://github.com/jenkinsci/mailer-plugin/compare/38c84e2320ee...c5497e08f35e" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/mailer-plugin/compare/38c84e2320ee...c5497e08f35e</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/slide_o_mix.png" title="slide_o_mix's avatar" /> slide_o_mix:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-23256?focusedId=227722&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-227722">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Any updates on this?</p>
</pre>
</details>

</ul>
</details>

<p>Any updates on this?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>davyboyhayes:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-23256?focusedId=227731&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-227731">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Alas, time and tide waits for no man, neither do companies. I was unable to dedicate time towards this, and our configuration has moved on beyond this where we no longer have Perforce jobs configured.</p>

<p>As I am unable to assist any further (and I'm the reporter) I'm happy to close the issue as Won't Do.</p>
</pre>
</details>

</ul>
</details>

<p>Alas, time and tide waits for no man, neither do companies. I was unable to dedicate time towards this, and our configuration has moved on beyond this where we no longer have Perforce jobs configured.</p>

<p>As I am unable to assist any further (and I'm the reporter) I'm happy to close the issue as Won't Do.</p>
